sycl/include/CL/sycl/detail/image_ocl_types.hpp

```diff
@@ -207,6 +206,7 @@ template <access::mode AccessMode> struct opencl_image1d_buffer_type;
 __SYCL_IMAGE_BUFFER_TY_DEFINE(read, ro);
 __SYCL_IMAGE_BUFFER_TY_DEFINE(write, wo);
 __SYCL_IMAGE_BUFFER_TY_DEFINE(discard_write, wo);
+__SYCL_IMAGE_BUFFER_TY_DEFINE(discard_read_write, rw);
```

Comment:
Why this change is needed? The comment above says `__SYCL_IMAGE_BUFFER_TY_DEFINE` is not used anymore.

Reply:
This change was needed a there was a specialization of accessor class with `discard_read_write`. Since we "unguard" this code from `-fsycl-explicit-simd` I need to add that.

Anyway, with my recent changes, this code is gone.